homeassistant/helpers/integration_platform.py

```diff
@@ -12,23 +13,35 @@
 from homeassistant.setup import ATTR_COMPONENT
 
 _LOGGER = logging.getLogger(__name__)
+DATA_INTEGRATION_PLATFORMS = "integration_platforms"
 
 
-@bind_hass
-async def async_process_integration_platforms(
-    hass: HomeAssistant,
-    platform_name: str,
-    # Any = platform.
-    process_platform: Callable[[HomeAssistant, str, Any], Awaitable[None]],
+@dataclass(frozen=True)
+class IntegrationPlatform:
+    """An integration platform."""
+
+    platform_name: str
+    process_platform: Callable[[HomeAssistant, str, Any], Awaitable[None]]
+    seen_components: set[str]
+
+
+async def async_process_integration_platform(
+    hass: HomeAssistant, component_name: str
 ) -> None:
-    """Process a specific platform for all current and future loaded integrations."""
+    """Process component being loaded or on demand."""
+    if "." in component_name:
+        return
 
-    async def _process(component_name: str) -> None:
-        """Process component being loaded."""
-        if "." in component_name:
-            return
+    integration = await async_get_integration(hass, component_name)
 
-        integration = await async_get_integration(hass, component_name)
+    integration_platforms: list[IntegrationPlatform] = hass.data[
+        DATA_INTEGRATION_PLATFORMS
+    ]
+    for integration_platform in integration_platforms:
+        platform_name = integration_platform.platform_name
```

Comment:
The code inside the loop should be its own function and we can use gather on it.  We can also use it below to avoid enumerating the other platforms

Reply:
Adjusted. Sorry for not noticing that sooner